### PR TITLE
fix: [CDS-47297]: removing id d from yaml for jenkins job parameters

### PR DESCRIPTION
**Pull request by satyam-harness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by Sagar-Rajput7 — APPROVED** [review]

File: src/modules/70-pipeline/components/PipelineSteps/Steps/JenkinsStep/JenkinsStepBase.tsx [hunk 1]
```diff
@@ -425,9 +423,9 @@ function FormContent({
                     <Spinner />
                   ) : (
                     isArray(formValues.spec.jobParameter) &&
-                    formValues.spec.jobParameter?.map(({ id }: jobParameterInterface, i: number) => {
+                    formValues.spec.jobParameter?.map((type: jobParameterInterface, i: number) => {
```
Comment (ishantmehtaharness) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/JenkinsStep/JenkinsStepBase.tsx:426]: On edit case, does it have a `type.id`?

Comment (ishantmehtaharness) [reply src/modules/70-pipeline/components/PipelineSteps/Steps/JenkinsStep/JenkinsStepBase.tsx:426]: Are we introducing id in Backend side?

**Review by pankajsingh-harness — APPROVED** [review]

**Comment by satyam-harness** [comment]
retrigger jest

File: src/modules/70-pipeline/components/PipelineSteps/Steps/JenkinsStep/JenkinsStepBase.tsx (hunk 1 above)
Comment (satyam-harness) [reply src/modules/70-pipeline/components/PipelineSteps/Steps/JenkinsStep/JenkinsStepBase.tsx:426]: yes

**Comment by satyam-harness** [comment]
wil increase coverage in this sprint -> ticket to track https://harness.atlassian.net/browse/PIE-6785

**Pull request merged**